src/language-js/embed/markdown.js

```diff
@@ -2,27 +2,44 @@
 
 const {
   builders: { indent, softline, literalline, dedentToRoot },
+  utils: { mapDoc },
 } = require("../../document/index.js");
-const { escapeTemplateCharacters } = require("../print/template-literal.js");
+const { printTemplateExpressions } = require("../print/template-literal.js");
 
+// The counter is needed to distinguish nested embeds.
+let counter = 0;
 function format(path, print, textToDoc) {
   const node = path.getValue();
-  let text = node.quasis[0].value.raw.replace(
-    /((?:\\\\)*)\\`/g,
-    (_, backslashes) => "\\".repeat(backslashes.length / 2) + "`"
-  );
+  const placeholder = `PRETTIER_MARKDOWN_PLACEHOLDER_${counter}_IN_JS`;
+  counter = (counter + 1) >>> 0;
+  let text = node.quasis
+    .map((quasi) => quasi.value.raw)
+    .join(placeholder)
+    .replace(
+      /((?:\\\\)*)\\`/g,
+      (_, backslashes) => "\\".repeat(backslashes.length / 2) + "`"
+    );
   const indentation = getIndentation(text);
   const hasIndent = indentation !== "";
   if (hasIndent) {
     text = text.replace(new RegExp(`^${indentation}`, "gm"), "");
   }
-  const doc = escapeTemplateCharacters(
+  const expressionDocs = printTemplateExpressions(path, print);
+  const doc = mapDoc(
     textToDoc(
       text,
       { parser: "markdown", __inJsTemplate: true },
       { stripTrailingHardline: true }
     ),
-    true
+    (currentDoc) =>
+      typeof currentDoc !== "string"
+        ? currentDoc
+        : currentDoc.split(placeholder).flatMap((unescaped, index, splits) => {
+            const escaped = unescaped.replace(/(\\*)`/g, "$1$1\\`");
```

Comment:
Can you create new function `escapeTemplateCharacter` in `../print/template-literal.js`?

```js
function escapeTemplateCharacter(unescapedText) {
  unescapedText.replace(/(\\*)`/g, "$1$1\\`");
}
```

And use it on this and `escapeTemplateCharacters`?

Reply:
You bet, but this function is coupled with: https://github.com/prettier/prettier/blob/6106421f0a7f9f241ae1de64fb22f0a21a9c0778/src/language-js/embed/markdown.js#L10-L13

They're reciprocals of each other (unescape backticks, escape backticks) and they're coupled with Markdown:
1. backticks are significant in Markdown, otherwise we wouldn't bother: just format the raw value like `\$` and other escape sequences
2. backticks can't exist in template literals, so `escapeBackticks(unescapeBackticks(text))` is non-destructive

So promoting either or both to `../print/template-literal.js` isn't needed?

What I have done now is:
- remove it from `../print/template-literal.js`
- extract both functions in `embed/markdown.js`
- in `embed/graphql.js` use `uncookTemplateElementValue()` directly